gulpfile.js

```diff
@@ -1141,6 +1142,13 @@ gulp.task("jsdoc", function (done) {
   });
 });
 
+gulp.task("dts", function (done) {
+  console.log();
+  console.log("### Generating typescript definitions");
+
+  exec("tsc -p .", done);
```

Comment:
Also, I do wonder if it'd make sense to define the configuration *inline* here rather than using a separate JSON file!?

For one thing, having the output directory of the `tsc` command controlled by an *external* file doesn't seem great when other gulp tasks (i.e. `dist-pre`) depend on being able to find the files is a particular spot.

Reply:
done